File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.py
def solution(n, s):
    # 집합의 합: n
    # 1 ~ n / 2까지, 합이 s
    
    # 1. n > s면 최고의 집합 존재 X
    if  n > s:
        return [-1]
    
    # 2. 곱셈이 커짐 -> 3,4 등...
    # s // n
    fir = s // n
    num = s % n
    
    answer = [fir] * n
    
    for i in range(num):
        answer[-1 - i] += 1
    
    return answer

File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.java
class Solution {
    public int[] solution(int n, int s) {
        // 집합의 합: s
        // 집합의 원소 개수: n

        // 1. n > s면 최고의 집합 존재 X
        if (n > s) {
            return new int[]{-1};
        }

        // 2. 곱이 최대가 되려면 최대한 균등하게 분배
        // s / n : 기본값
        int fir = s / n;
        int num = s % n;

        int[] answer = new int[n];

        for (int i = 0; i < n; i++) {
            answer[i] = fir;
        }

        for (int i = 0; i < num; i++) {
            answer[n - 1 - i]++;
        }

        return answer;
    }
}